### ISSUE TITLE
Not compatible with Python >=3.11

### ISSUE THREAD

**Issue by mhechthz** [p5py/p5#466]
On windows the library is not compatible with Python 3.11 since Pillow 9.0.x is not compatible with Python 3.11. Compilation is needed which is never a good idea since it failes on most systems. Python 3.10 works well but is outdated now.

**Comment by hoarffrost** [comment]
@mhechthz can we not change Pillow to its latest version 10.1.0 which is compatible with python 3.11 ? It worked for me so thats why I am suggesting. And if there is any issue with that let me know as I am new to open Source ✌️.

**Comment by mhechthz** [comment]
Yes @Gyan313, this is exactly what I would like to be changed. So I can directly install the library using pip instead of compiling it on my own machine during installation (which never works on many windows machines and for many colleagues I know).

**Comment by tushar5526** [comment]
Thanks, @mhechthz @Gyan313 for testing it out on windows. I should probably add a basic GHA for windows that installs the p5 on it. Would any of you like to make a PR to bump the Pillow dependency?

**Comment by hoarffrost** [comment]
@tushar5526 Hi, I'm also working on integrating a pre-commit hook to p5, and I'm currently addressing some flake8 issues. Once I'm finished, I'll submit a pull request to update the Pillow dependency. Whether or not I resolve the flake8 problem, I'll still create a PR for Pillow today. ✌️❤️

**Comment by tushar5526** [comment]
Hey @Gyan313 awesome to hear about your plans! Just a heads-up on something related to pre-commit already in the pipeline - https://github.com/p5py/p5/pull/460

**Comment by leoyu037** [comment]
This is also a problem on macOS with intel processors

**Comment by 8ude** [comment]
Still getting this issue when trying to install p5 with pip. Python version 3.11.9 running on Windows 11 v10.0.22631

I was able to install and run with Python 3.10.11

**Comment by jfk02** [comment]
I have python 3.13.1 on Windows 10. I have installed the p5 library, which is working so far. I installed the dependencies separately:

pip install numpy
pip install pillow
pip install pyopengl
pip install glfw
pip install vispy
pip install skia-python

pip install --no-deps p5

all in the latest versions.

**Comment by swirly** [comment]
Well, just tried a simple thing. Modified the `requirements.txt` to replace pil and vispy version from = to >=.
Installed p5 with the local directory with : 

```bash
pip install .
```

And it works flawlessly in python 3.13

The correction should not be complicated : just allow superior version, not just equal version in requirement.txt. That's it !

Issue could be resolved really quick.
